HealthGPT/HealthGPT/HealthDataInterpreter.swift

```diff
@@ -0,0 +1,51 @@
+//
+// This source file is part of the Stanford HealthGPT project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University & Project Contributors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Spezi
+import SpeziFHIR
+import SpeziOpenAI
+import Foundation
+
+class HealthDataInterpreter<ComponentStandard: Standard>: DefaultInitializable, Component, ObservableObject, ObservableObjectProvider {
+    required init() {}
+
+    func generateMainPrompt() async throws -> [Chat] {
+        let healthDataFetcher = HealthDataFetcher()
+        let healthData = try await healthDataFetcher.fetchAndProcessHealthData()
+
+        let generator = PromptGenerator(with: healthData)
+        let mainPrompt = generator.buildMainPrompt()
+        return [Chat(role: .system, content: mainPrompt)]
+    }
+    
+    func queryPrompt(oldPrompt: [Chat], newMessage: [Chat]) async throws -> [Chat] {
```

Comment:
It would probably make sense to change this function in a way that it is called when the internal chat state changes (as referenced and noted above). This results in the fact that you only have a single source of truth and you will not need to combine two different chat instances in a single chat.